pkg/operation/botanist/component/nginxingressshoot/nginxingress.go

```diff
@@ -624,6 +616,13 @@ func (n *nginxIngress) computeResourcesData() (map[string][]byte, error) {
 		}
 	}
 
+	if n.values.KubeAPIServerHost != nil {
+		deploymentController.Spec.Template.Spec.Containers[0].Env = append(deploymentController.Spec.Template.Spec.Containers[0].Env, corev1.EnvVar{
+			Name:  "KUBERNETES_SERVICE_HOST",
+			Value: *n.values.KubeAPIServerHost,
+		})
```

Comment:
Shouldn't this be injected by the webhook for all pods?

Reply:
Oh, I was not aware of this. We only compared the deployment yesterday, and saw this was missing.

Reply:
I am not sure why this was in the chart earlier, so probably it makes sense to double-check this assumption.

Reply:
In my local setup, the pods have this variable added even though I removed it from the deployment.